sdk/servicebus/service-bus/src/sender.ts

```diff
@@ -199,27 +208,53 @@ export class SenderImpl implements Sender {
     return this._isClosed || this._context.isClosed;
   }
 
-  async send(message: ServiceBusMessage, options?: OperationOptions): Promise<void> {
+  async send(message: ServiceBusMessage, options?: OperationOptions): Promise<void>;
+  async send(messages: ServiceBusMessage[], options?: OperationOptions): Promise<void>;
+  async send(messageBatch: ServiceBusMessageBatch, options?: OperationOptions): Promise<void>;
+  async send(
+    messageOrMessagesOrBatch: ServiceBusMessage | ServiceBusMessage[] | ServiceBusMessageBatch,
+    options?: OperationOptions
+  ): Promise<void> {
     this._throwIfSenderOrConnectionClosed();
-    throwTypeErrorIfParameterMissing(this._context.namespace.connectionId, "message", message);
-    return this._sender.send(message, options);
+
+    if (Array.isArray(messageOrMessagesOrBatch)) {
+      const batch = await this.createBatch(options);
+
+      for (const message of messageOrMessagesOrBatch) {
+        if (!batch.tryAdd(message)) {
+          // this is too big - throw an error
+          const error = new MessagingError(
+            "Messages were too big to fit in a single batch. Remove some messages and try again or create your own batch using createBatch(), which gives more fine-grained control."
+          );
+          error.code = "MessageTooLargeError";
+          throw error;
+        }
+      }
+
+      return this.send(batch, options);
+    } else if (isServiceBusMessageBatch(messageOrMessagesOrBatch)) {
+      this._throwIfSenderOrConnectionClosed();
+      throwTypeErrorIfParameterMissing(
+        this._context.namespace.connectionId,
+        "messageBatch",
+        messageOrMessagesOrBatch
```

Comment:
If we are here, then it means `isServiceBusMessageBatch()` returns true. If that is so, then what is the point of checking whether parameter is missing? We already know we got the batch right?

Reply:
Removed _and_ added in a unit test (I realized I was missing the undefined/null case).